### PR TITLE
feat: handle unresolved values

### PR DESCRIPTION
**Pull request by JesperBllnbm**
## Checklist

- [x] My PR title is in the style of [conventional commits](https://www.conventionalcommits.org/)
- [x] All public facing APIs are documented with [dartdoc](https://dart.dev/guides/language/effective-dart/documentation)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: lib/src/domain/models/tokens_by_file_type/tokens_by_type.dart [hunk 1]
```diff
@@ -22,4 +23,31 @@ sealed class TokensByType with _$TokensByType {
   }) = _TokensByType;
 
   const TokensByType._();
+
+  /// A map of name and [DesignToken] iterable containing all tokens.
+  Map<String, Iterable<DesignToken>> get allTokens => {
```
Comment (JesperBllnbm) [inline lib/src/domain/models/tokens_by_file_type/tokens_by_type.dart:28]: Rename to tokens as map

File: lib/src/domain/providers/design_token_providers.dart [hunk 2]
```diff
@@ -8,8 +9,54 @@ import 'package:figmage/src/domain/providers/logger_providers.dart';
 import 'package:figmage/src/domain/repositories/styles_repository.dart';
 import 'package:figmage/src/domain/repositories/variables_repository.dart';
 import 'package:figmage/src/domain/util/token_filter_x.dart';
+import 'package:mason_logger/mason_logger.dart';
 import 'package:riverpod/riverpod.dart';
 
+/// Query for [filterUnresolvedTokensProvider]
+typedef FilterUnresolvedQuery = ({
```
Comment (JesperBllnbm) [inline lib/src/domain/providers/design_token_providers.dart:16]: Just settings as argument for family

File: lib/src/domain/util/token_filter_x.dart [hunk 3]
```diff
@@ -11,14 +11,42 @@ extension TokenFilterX<X> on Iterable<DesignToken<X>> {
...
-  /// Turns the variables into a map of values by name by mode
-  ///
-  // TODO(tim): this is in the repo I know but I didn't get how it works
-  Map<String, Map<String, X>> get valuesByNameByMode {
-    final sorted = sortedBy((element) => element.name);
-    final allModes = expand((token) => token.valuesByModeName.keys)
-        .toSet()
-        .sortedBy((mode) => mode);
+  /// Sorts tokens by their name property.
+  List<DesignToken<X>> sortTokensByName() {
```
Comment (JesperBllnbm) [inline lib/src/domain/util/token_filter_x.dart:15]: sortedByName

**Review by timcreatedit — APPROVED** [review]

**Review by timcreatedit — APPROVED** [review]
subber gmacht

**Pull request merged**